### PR TITLE
More docs on VPPs and reporters

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #22662

### CODE REVIEW

**Comment by cticenhour** [comment]
@GiudGiud I see you added another commit after your review request - is this ready to go now?

**Comment by GiudGiud** [comment]
ah yes. I was going to make a separate PR for fixups but did not find the time to add more so it's just this solo commit

File: framework/doc/content/source/vectorpostprocessors/NodalValueSampler.md [hunk 1]
```diff
@@ -5,6 +5,11 @@
 This `VectorPostprocessor` is used for sampling nodal variables at each node
 in the domain, selection of blocks, or selection of boundaries.
 
+!alert note title=Vector names / CSV output column names
+The `NodalValueSampler` declares a vector for each spatial coordinate, (`x`, `y`, `z`) of the sampled nodes,
+the ids of the nodes in a vector named `id`
+and a vector named after each variable sampled, containing the variable values at each point.
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/NodalValueSampler.md:11]:
```suggestion
`NodalValueSampler` declares a vector for each spatial coordinate, (`x`, `y`, `z`), of the sampled nodes,
the IDs of the nodes in a vector named `id`,
and a vector named after each variable sampled, containing the variable values at each point.
```

File: framework/doc/content/source/vectorpostprocessors/PointValueSampler.md [hunk 2]
```diff
@@ -5,7 +5,8 @@
 !alert note
 If the point value sampler is used with a discontinuous variable on the edge/face of a 2D/3D element, then the value from the element with the lowest ID will be returned.
 
-The CSV output, with rows for each sampled point, contains the following columns:
+The CSV output, with rows for each sampled point, contains the columns listed below. The vectors declared by the `PointValueSampler`
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/PointValueSampler.md:8]:
```suggestion
The CSV output, with rows for each sampled point, contains the columns listed below. The vectors declared by `PointValueSampler`
```

File: framework/doc/content/source/vectorpostprocessors/SideValueSampler.md [hunk 3]
```diff
@@ -2,18 +2,15 @@
 
 !syntax description /VectorPostprocessors/SideValueSampler
 
-The data output to CSV is output with the following columns:
+The data output to CSV is output with the columns listed below. The declared vectors use the same
+names.
 
 - the id of the element the quadrature point, on the side, belongs to
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/SideValueSampler.md:8]:
```suggestion
- the id of the element that the quadrature point, on the side, belongs to
```

Adjusting this while we're here

File: framework/doc/content/source/vectorpostprocessors/SideValueSampler.md [hunk 4]
```diff
@@ -2,18 +2,15 @@
...
 
-- the values of the requested variables
+- the values of the requested variables, with a column named after the variable
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/SideValueSampler.md:10]:
```suggestion
- the values of the requested variables, with a column named after each variable
```

File: framework/doc/content/source/vectorpostprocessors/SidesetInfoVectorPostprocessor.md [hunk 5]
```diff
@@ -4,12 +4,16 @@
...
 
-SidesetInfoVectorPostprocessor assembles information from sidesets and prints them to
+`SidesetInfoVectorPostprocessor` assembles information from sidesets and prints them to
 a csv file. Currently, it allows to obtain `area`, `centroid`, and the bounding box
 via `min` and `max`. `min` contains the minimum x, y, z coordinates of the
 sideset, while `max` contains the respective maximum values. Note that `centroid`
 is not guaranteed to be a point contained in the sideset.
 
+!alert note title=Vector names / CSV output column names
+The `SidesetInfoVectorPostprocessor` uses the following names for the vectors / CSV column headers
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/SidesetInfoVectorPostprocessor.md:14]:
```suggestion
`SidesetInfoVectorPostprocessor` uses the following names for the vectors / CSV column headers: 
```

File: framework/doc/content/source/vectorpostprocessors/SpatialUserObjectVectorPostprocessor.md [hunk 6]
```diff
@@ -17,6 +17,9 @@ provided, this object will attempt to get the spatial points directly from the u
 These points will represent locations in space where the user object obtains a unique value.
 To use this feature, the user object must define the `spatialPoints()` interface.
 
+!alert note title=Vector names / CSV output column names
+The `SpatialUserObjectVectorPostprocessor` declares a vector with its own name. The full reporter name ends up being `<vpp_object_name>/<vpp_object_name>`.
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/SpatialUserObjectVectorPostprocessor.md:21]:
```suggestion
`SpatialUserObjectVectorPostprocessor` declares a vector with its own name. The full reporter name ends up being `<vpp_object_name>/<vpp_object_name>`.
```

File: framework/doc/content/source/vectorpostprocessors/SphericalAverage.md [hunk 7]
```diff
@@ -13,6 +13,9 @@ The CSV data output consists of the following columns, ordered by column name:
 
 - average of inner and outer radius of each shell of the sphere
 
+!alert note title=Vector names / CSV output column names
+The `SphericalAverage` declares a vector for the average shell radii, named `radius` and a vector for each variable requested, named with the variable name.
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/SphericalAverage.md:17]:
```suggestion
`SphericalAverage` declares a vector for the average shell radii, named `radius`, and a vector for each variable requested, named with the variable name.
```

File: framework/doc/content/source/vectorpostprocessors/VariableValueVolumeHistogram.md [hunk 8]
```diff
@@ -6,10 +6,11 @@ This VectorPostprocessor lets you tabulate the volumes in the simulation domain
 
 The CSV output contains the two columns for:
 
-- the variable value bins upper values, with the name of the variable in the first row
+- the variable value bins upper values, with the name of the variable in the first row (header)
 
 - the volume in which the variable holds the values within the bins, with the name `n` in the first row
 
+The names of the vectors declared by the `VariableValueVolumeHistogram` match the CSV output column header names.
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/VariableValueVolumeHistogram.md:13]:
```suggestion
The names of the vectors declared by `VariableValueVolumeHistogram` match the CSV output column header names.
```

File: framework/doc/content/source/vectorpostprocessors/VectorOfPostprocessors.md [hunk 9]
```diff
@@ -8,6 +8,8 @@ This vector postprocessor is mainly used for two purposes:
 
 - to combine postprocessors to match the expected vector postprocessor input of a kernel, a user object, etc
 
+!alert note title=Vector names / CSV output column names
+The `VectorOfPostprocessors` declares a vector with its own name. The full reporter name ends up being `<vpp_object_name>/<vpp_object_name>`.
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/VectorOfPostprocessors.md:12]:
```suggestion
`VectorOfPostprocessors` declares a vector with its own name. The full reporter name ends up being `<vpp_object_name>/<vpp_object_name>`.
```

File: framework/doc/content/source/vectorpostprocessors/WorkBalance.md [hunk 10]
```diff
@@ -10,6 +10,9 @@ The idea here is to compute per-processor metrics to help in determining the qua
 
 Currently computes: number of local elements, nodes, dofs and partition sides.  The partition sides are the sides of elements that are on processor boundaries (also known as the "edge-cuts" in partitioner lingo).  Also computes the "surface area" of each partition (physically, how much processor boundary each partitioning has).
 
+!alert note title=Vector names
+The names of the vectors declared by the `WorkBalance` are the names of [!param](/VectorPostprocessors/WorkBalance/balances) requested.
```
Comment (cticenhour) [inline framework/doc/content/source/vectorpostprocessors/WorkBalance.md:14]:
```suggestion
The names of the vectors declared by `WorkBalance` are the names of the [!param](/VectorPostprocessors/WorkBalance/balances) requested.
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
Thanks for the clean-up and adjustments here! A few suggestions and corrections to note.

**Comment by GiudGiud** [comment]
applied all suggested changes!
had to squash the two commits I did for that

**Review by cticenhour — APPROVED** [review]

**Pull request merged**